packages/protocol/contracts/governance/DowntimeSlasherIntervals.sol

```diff
@@ -0,0 +1,307 @@
+pragma solidity ^0.5.3;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "./SlasherUtil.sol";
+
+contract DowntimeSlasherIntervals is SlasherUtil {
+  using SafeMath for uint256;
+
+  // Maps validator address -> epoch number -> a block interval for which the
+  // validator has been slashed.
+  mapping(address => mapping(uint256 => uint256[2])) slashedIntervals;
```

Comment:
I'm trying to think of ways to simplify this a bit. What if, instead, we store just the `endBlock` of the last interval for which each validator was slashed, and require that the `startBlock` of the next interval for which the validator was slashed is greater than this value?

Reply:
@asaj 
The problem that I found there is that if you have a validator that was down 2 epochs in a row, one slash in the last epoch, will "save" the validator from the older epoch slash.

Is there any chance that is something that will happen? I don't know, but we are giving a 4 epoch time to seek for these kind of errors.

Other thing that we could do, is to calculate the startBlock from the endBlock and the slashableDowntime, but if for some reason via governance, we decide to change the slashableDowntime, we will be facing some other issues, like new slashes in intervals that were already slashed